apis/v1alpha1/gateway_types.go

```diff
@@ -77,6 +77,10 @@ type GatewaySpec struct {
 	// As a special case, each group of compatible listeners
 	// may contain exactly one Listener with a match type of "Any".
 	//
+	// Listeners across Gateway resources cannot be collapsed i.e. Gateway
+	// merging is not allowed. Each Gateway resource is a distinct network
+	// endpoint.
```

Comment:
Since you are soliciting suggestions on the wording:
```suggestion
	// Listeners in different Gateways cannot be collapsed.  Each Gateway
	// resource represents a distinct logical network endpoint.
```
And where we talk about collapsing listeners above, we could state more explicitly that they can be collapsed only _within a single gateway_:
```go
	// In general, each listener in this array must have a unique port.
	// However, an implementation may collapse listeners within the same
	// Gateway definition that have the same port into a single listener,
	// as long as the listeners are "compatible", as defined below.
```